### PR TITLE
Allowed users to expand/hide search facets by clicking facet title

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/2120

#### What's this PR do?
- Made facets title clickable, user can expand/hide search facets by clicking facet title

#### How should this be manually tested?
- Go to learners page and click the title of any facet.

@pdpinch 
#### Screenshots (if appropriate)
screen shot 2017-03-06 at 6 28 34 pm

### CODE REVIEW

**Comment by aliceriot** [comment]
works great! looking at code now.

File: static/js/components/LearnerSearch.js [hunk 1]
```diff
@@ -200,58 +200,63 @@ export default class LearnerSearch extends SearchkitComponent {
       <Card className="fullwidth" shadow={1}>
         <FilterVisibilityToggle
           {...this.props}
+          title="Course"
           filterName="courses"
         >
           <HierarchicalRefinementFilter
             field={"program.enrollments"}
-            title="Course"
+            title=""
```
Comment (aliceriot) [inline static/js/components/LearnerSearch.js]: can we just omit the `title` prop from these filters, or do we have to pass in the empty string?

File: static/js/components/search/FilterVisibilityToggle.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ export const FILTER_ID_ADJUST = {
 
 export default class FilterVisibilityToggle extends SearchkitComponent {
   props: {
+    title:             string,
```
Comment (aliceriot) [inline static/js/components/search/FilterVisibilityToggle.js]: can you fix the alignment for this prop? (with the indentation of the other props, just for readability)

File: static/js/components/search/FilterVisibilityToggle.js [hunk 3]
```diff
@@ -71,6 +72,20 @@ export default class FilterVisibilityToggle extends SearchkitComponent {
...
+    }
+    const { title } = this.props;
+    return (
+      <div
+        className="sk-hierarchical-refinement-list__header"
+        onClick={this.toggleFilterVisibility}>
+        {title}
+      </div>
+    );
+  };
```
Comment (aliceriot) [inline static/js/components/search/FilterVisibilityToggle.js]: why not roll this into the `openStateIcon` method above? these two methods basically do the same thing (check `this.isInResults` and if so return some UI, else null).

You could write a new method like this:

```js
renderFilterTitle = (children) => {
  if (!this.isInResults(children.props.id)) {
    return null;
  }
  const { title } = this.props;
  return <div className="title-row" onClick={this.toggleFilterVisibility}>
    <Icon
      name="arrow_drop_down"
      className={this.openClass()}
    />
    <div className="sk-hierarchical-refinement-list__header">       <div className="sk-hierarchical-refinement-list__header">
      { title } 
    </div>
  </div>;
}
```

something along those lines! then we're just making the check once, only setting one `onClick` handler, etc.
             
what do you think?

File: static/js/components/LearnerSearch.js (hunk 1 above)
Comment (amir-qayyum-khan) [inline static/js/components/LearnerSearch.js]: they are compulsory by these components

Comment (aliceriot) [inline static/js/components/LearnerSearch.js]: gotcha, that's too bad. ah well.

**Comment by amir-qayyum-khan** [comment]
@aliceriot any update?

**Comment by noisecapella** [comment]
@amir-qayyum-khan She's out today

**Comment by amir-qayyum-khan** [comment]
ok @noisecapella

**Pull request merged**